openpgp/read.go

```diff
@@ -336,7 +344,14 @@ func (scr *signatureCheckReader) Read(buf []byte) (n int, err error) {
 
 		var ok bool
 		if scr.md.Signature, ok = p.(*packet.Signature); ok {
-			scr.md.SignatureError = scr.md.SignedBy.PublicKey.VerifySignature(scr.h, scr.md.Signature)
+			var err1, err2 error
+			if fingerprint := scr.md.Signature.IssuerFingerprint; fingerprint != nil {
+				if !hmac.Equal(fingerprint, scr.md.SignedBy.PublicKey.Fingerprint[:]) {
+					err1 = errors.StructuralError("bad key fingerprint")
+				}
+			}
+			err2 = scr.md.SignedBy.PublicKey.VerifySignature(scr.h, scr.md.Signature)
```

Comment:
Any reason not to do an `if err != nil` here, rather than inventing the `pickFirstNonNilError` concept? If we want we could even factor this section out into a separate function, and then that function could use the usual `if err != nil { return err }` short circuit behavior, to avoid nesting too many if's. Not necessary here, but maybe if there was ever a third error?